### PR TITLE
[sival,alert] Enable `alert_handler_lpg_sleep_mode_pings` test in sival environments

### PR DESCRIPTION
**Pull request by jwnrt**
* Added the sival test environments, but kept Verilator disabled.
* Changed the platform-dependent handling to a switch/case to make the verilator case clearer.
* Added a case for silicon.

### CODE REVIEW

File: sw/device/tests/alert_handler_lpg_sleep_mode_pings.c [hunk 1]
```diff
@@ -80,23 +80,36 @@ static void init_peripherals(void) {
...
+      busy_spin_micros(1000 * 200);
+      break;
+    case kDeviceSilicon:
+      // 2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+      // 2*4*(2**16)*(42ns) = 22ms
+      busy_spin_micros(1000 * 22);
+      break;
+    case kDeviceSimDV:
+      // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+      // 2*4*(2**3)*(40ns) = 3us
```
Comment (moidx) [inline sw/device/tests/alert_handler_lpg_sleep_mode_pings.c:97]: Shouldn't this be `(2**16)`? The clock and RTL params configuration between `kDeviceSilicon` and `kDeviceSimDV` should be the same.

Comment (jwnrt) [reply sw/device/tests/alert_handler_lpg_sleep_mode_pings.c]: Updated

**Comment by jwnrt** [comment]
Failing on the FPGA because of https://github.com/lowRISC/opentitan/issues/23038

**Comment by jwnrt** [comment]
Closed because it’s against an old branch, will need redoing on `earlgrey_1.0.0` when the splicing issue is fixed.

**Pull request closed without merging**